shared/utility/math/matrix/transform.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2013 NUbots <nubots@nubots.net>
+ */
+
+#ifndef UTILITY_MATH_TRANSFORM_H
+#define UTILITY_MATH_TRANSFORM_H
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+#include "utility/math/angle.h"
+
+/**
+ * Matrix related helper methods
+ *
+ * @author Brendan Annable
+ */
+
+namespace utility {
+namespace math {
+    namespace transform {
+
+        inline Eigen::Affine2d localToWorld(const Eigen::Affine2d& local, const Eigen::Affine2d& reference) {
+            // translates to this + rotZ(this.angle) * reference
+            Eigen::Rotation2Dd R(local.linear());
+            Eigen::Vector2d newPos = R * reference.translation();
+            Eigen::Affine2d result;
+            result.translation() = local.translation() + newPos;
+            // do not use normalizeAngle here, causes bad things when turning! TODO: unsure on cause
+            result.linear() = Eigen::Rotation2Dd(Eigen::Rotation2Dd(local.linear()).angle()
+                                                 + Eigen::Rotation2Dd(reference.linear()).angle())
+                                  .toRotationMatrix();
+            return result;
+        }
```

Comment:
Don't the Affine2d already do this by multiplying them together?

Reply:
will have to look into that